src/lib/party.ts

```diff
@@ -0,0 +1,784 @@
+    this.lastVrfOutput_.check = new Bool(true);
+
+    if (this.lastVrfOutput_.value === null) {
+      throw new Error('Cannot get lastVrfOutput before it was set.');
+    } else {
+      return this.lastVrfOutput_.value;
+    }
+  }
+}
+
+/**
+ * Ignores a `dummy`
+ *
+ * @param dummy The value to ignore
+ * @returns Always an ignored value regardless of the input.
+ */
+function ignore<A>(dummy: A): OrIgnore<A> {
+  return new OrIgnore(new Bool(false), dummy);
+}
+/*
+function check<A>(dummy: A): OrIgnore<A> {
+  return new OrIgnore(new Optional(new Bool(true), dummy));
+} */
+
+/**
+ * Ranges between all uint32 values
+ */
+const uint32 = () => new ClosedInterval(UInt32.fromNumber(0), UInt32.MAXINT());
+
+/**
+ * Ranges between all uint64 values
+ */
+const uint64 = () => new ClosedInterval(UInt64.fromNumber(0), UInt64.MAXINT());
+
+export class AccountPredicate {
+  balance: ClosedInterval<UInt64>;
+  nonce: ClosedInterval<UInt32>;
+  receiptChainHash: OrIgnore<Field>;
+  publicKey: OrIgnore<PublicKey>;
+  delegate: OrIgnore<PublicKey>;
+  state: Array<OrIgnore<Field>>;
+  sequenceState: OrIgnore<Field>;
+  provedState: OrIgnore<Bool>;
+
+  static ignoreAll(): AccountPredicate {
+    let appState: Array<OrIgnore<Field>> = [];
+    for (let i = 0; i < SnappStateLength; ++i) {
+      appState.push(ignore(Field.zero));
+    }
+
+    return new AccountPredicate(
+      uint64(),
+      uint32(),
+      ignore(Field.zero),
+      ignore(new PublicKey(Group.generator)),
+      ignore(new PublicKey(Group.generator)),
+      appState,
+      ignore(Field.zero),
+      ignore(new Bool(false))
+    );
+  }
+
+  constructor(
+    balance: ClosedInterval<UInt64>,
+    nonce: ClosedInterval<UInt32>,
+    receiptChainHash: OrIgnore<Field>,
+    publicKey: OrIgnore<PublicKey>,
+    delegate: OrIgnore<PublicKey>,
+    state: Array<OrIgnore<Field>>,
+    sequenceState: OrIgnore<Field>,
+    provedState: OrIgnore<Bool>
+  ) {
+    this.balance = balance;
+    this.nonce = nonce;
+    this.receiptChainHash = receiptChainHash;
+    this.publicKey = publicKey;
+    this.delegate = delegate;
+    this.state = state;
+    this.sequenceState = sequenceState;
+    this.provedState = provedState;
+  }
+}
+
+export class PartyBalance {
+  private body: Body;
+  constructor(body: Body) {
+    this.body = body;
+  }
+
+  addInPlace(x: Int64 | UInt32 | UInt64) {
+    this.body.delta = this.body.delta.add(x);
+  }
+
+  subInPlace(x: Int64 | UInt32 | UInt64) {
+    this.body.delta = this.body.delta.sub(x);
+  }
+}
+
+export class Party<P> {
+  body: Body;
+  predicate: P;
+
+  constructor(body: Body, predicate: P) {
+    this.body = body;
+    this.predicate = predicate;
+  }
+
+  get balance(): PartyBalance {
+    return new PartyBalance(this.body);
+  }
+
+  get update(): Update {
+    return this.body.update;
+  }
+
+  get publicKey(): PublicKey {
+    return this.body.publicKey;
+  }
+
+  static createUnsigned(publicKey: PublicKey): Party<void> {
```

Comment:
There's usually no reason to type-annotate function return values. TS infers these types for us!